presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergUtil.java

```diff
@@ -725,7 +608,10 @@ public static Object deserializePartitionValue(Type type, String valueString, St
             if (type.equals(DOUBLE)) {
                 return parseDouble(valueString);
             }
-            if (type.equals(DATE) || type.equals(TIME) || type.equals(TIMESTAMP)) {
+            if (type.equals(TIMESTAMP) || type.equals(TIME)) {
+                return MICROSECONDS.toMillis(parseLong(valueString));
+            }
```

Comment:
I noticed this earlier when authoring the histogram PR. Iceberg specifies TIME and TIMESTAMP as microsecond precision. However, while we might be able to read/write partition metadata with the correct values here, there is definitely issues with inter-engine interop here.

If you write a `TIME[STAMP]` field in Spark to an Iceberg table, I believe it should be microsecond precision. But if you write it with Presto, won't it write with millisecond-precision? I haven't done much investigation into this but it seems like we could be inadvertently writing incompatible data with Presto if we don't resolve the data to microseconds.

I don't think it's something we need to fix in this PR, but it may be worth updating the supported types to only include `TIMESTAMP_MICROSECONDS` and similar instead of standard `TIMESTAMP` for Iceberg tables..

Reply:
Agree, currently the TimestampType/TimeType indeed bring a lot inconvenient. We really need to start considering and updating the supported type in Iceberg to consist with the spec.